src/utils/presence.js

```diff
@@ -39,20 +39,30 @@ export const getAggregatedPresence = (presence: UserPresence): ClientPresence =>
    * If there are several ClientPresence objects with the greatest
    * PresenceStatus, an arbitrary one is chosen.
    */
+
+  let status = 'offline';
+  let client = '';
+  let timestamp = 0;
+
   Object.keys(presence)
-    .filter((client: string) => client !== 'aggregated')
-    .reduce(
-      (aggregated, client: string) => {
-        const { status, timestamp } = presence[client];
-        if (Date.now() / 1000 - timestamp < OFFLINE_THRESHOLD_SECS) {
-          if (presenceStatusGeq(status, aggregated.status)) {
-            return { client, status, timestamp };
-          }
+    .filter((c: string) => c !== 'aggregated')
+    .forEach(c => {
+      const devicePresence = presence[c];
+      const age = Date.now() / 1000 - devicePresence.timestamp;
```

Comment:
One followup thought: `Date.now()` should be called once at the top of the loop, like `const now = Date.now()`. That way the whole function is working from the same value of "now", which helps keep it sane to reason about.